### PR TITLE
Daily job builds azure plugins 

### PR DESCRIPTION
**Pull request by 0xmichalis**
@openshift/sig-azure this PR adds a build job for the sync plugin using the ci-operator. WIP until I actually make it work.

### CODE REVIEW

File: ci-operator/config/openshift/azure-helm/master.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "base_images": {
+    "base": {
+      "cluster": "https://api.ci.openshift.org",
+      "namespace": "openshift",
+      "name": "origin-v3.11",
+      "tag": "base"
+    }
+  },
+
+  "canonical_go_repository": "github.com/jim-minter/azure-helm",
```
Comment (mjudeikis) [inline ci-operator/config/openshift/azure-helm/master.json]: Should we wait for migration?

**Comment by kargakis** [comment]
I've got glide working, trying to fix a weird compilation failure
```
rm -f sync tunnel
go generate ./...
bindata.go
bindata.go
bindata.go
CGO_ENABLED=0 go build ./cmd/sync
pkg/addons/clean.go:10:2: 
pkg/jsonpath/parser.go:2:1: expected 'package', found 'import'
```

**Comment by kargakis** [comment]
Seems like `go generate ./...` breaks the content of `parser.go` ... Trying to figure out why..

File: ci-operator/config/openshift/azure-helm/master.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "test_base_image": {
+    "cluster": "https://api.ci.openshift.org",
+    "namespace": "azure",
+    "name": "glide",
+    "tag": "go1.10"
+  },
+
+  "canonical_go_repository": "github.com/jim-minter/azure-helm",
+
+  "binary_build_commands": "glide install -v; pushd vendor/github.com/Azure/acs-engine; make generate; popd; make sync",
```
Comment (jim-minter) [inline ci-operator/config/openshift/azure-helm/master.json]: why do we do this?

**Comment by jim-minter** [comment]
the above compilation failure fixed by https://github.com/golang/tools/commit/4735f53fb37db012e63d1a62914417980fb3c023

**Comment by kargakis** [comment]
@jim-minter this should be ready to get merged. I am going to follow-up with mirroring to our quay repos in a different PR.

File: ci-operator/config/openshift/azure-helm/master.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{
+  "tag_specification": {
```
Comment (jim-minter) [inline ci-operator/config/openshift/azure-helm/master.json]: @kargakis if you happen to have a link to where this json spec is defined/documented please can you send it my way?

Comment (mjudeikis) [reply ci-operator/config/openshift/azure-helm/master.json]: aos-azure way

Comment (kargakis) [reply ci-operator/config/openshift/azure-helm/master.json]: No docs unfortunately atm but they are underway (cc @petr-muller @stevekuznetsov). I can demo what I know in a scrum call or during demo day.

Comment (jim-minter) [reply ci-operator/config/openshift/azure-helm/master.json]: possibly https://github.com/openshift/ci-operator/blob/master/pkg/api/types.go is what I was looking for in the shortest term

**Pull request merged**